test/unit/server_test.js

```diff
@@ -1117,5 +1117,56 @@ describe("server.js tests", function () {
           })
       });
     });
+
+    describe("TradeAggregationCallBuilder", function () {
+      let tradeAggregationResponse = +{
+        "_links": {
+        "self": {
+                "href": "https://horizon.stellar.org/trade_aggregations?base_asset_type=native\u0026start_time=1512689100000\u0026counter_asset_issuer=GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH\u0026limit=200\u0026end_time=1512775500000\u0026counter_asset_type=credit_alphanum4\u0026resolution=300000\u0026order=asc\u0026counter_asset_code=BTC"
+                },
+            "next": {
+        "href": "https://horizon.stellar.org/trade_aggregations?base_asset_type=native\u0026counter_asset_code=BTC\u0026counter_asset_issuer=GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH\u0026counter_asset_type=credit_alphanum4\u0026end_time=1512775500000\u0026limit=200\u0026order=asc\u0026resolution=300000\u0026start_time=1512765000000"
+              }
+          },
+        "_embedded": {
+          "records": []
+        }
+      } 
```

Comment:
I have to admit that I don't fully understand this pattern. If all that's being tested is the generated URL, why do we mock a real shaped response? Wouldn't this work just as well with: 
```javascript
tradeAggregationResponse = "bla"
```

Reply:
Yes you are right, this should work as well with some arbitrary data. This is just included should we want to extend the tests to cover more scenarios, like response structure. At the moment, what this just does is to ensure that the function is being called in the right format.

Reply:
IMHO, mocks should return data similar to real data. For example, we want to test if SDK creates functions for `_links`.